src/glfw-sandbox/engine.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+#include <windows.h>
+#include <GL/gl.h>
+#include <GL/glu.h>
+
+#define GLFW_INCLUDE_NONE
+#include <GLFW/glfw3.h> // Will give use GLFW + GL
+
+#include <iostream>
+
+
+class windowCreator
+{
+private:
+	int resWidth{};
+	int resHeight{};
+	
+	// Could you please clarify what these three do?
+	// Are these ponters? If posssible just a little clarification how this works
+	// This might be tied to main.cpp line 23-32
+	char *title;
+	GLFWmonitor *monitor;
+	GLFWwindow *share;
+
+	int fieldOfView{};
+
+
+public:
+
+	windowCreator(int resW, int resH, int fov) // for now leave other values and hardcode like a noob, Petr help <3 :D
+		: resWidth{ resW }, resHeight{ resH }, fieldOfView{fov}
+	{	
+		if (!glfwInit())
+		{
+			std::cout << "Failed to initialize GLFW library" << std::endl;
+			//return 1; Why I can't return 1 here?
+		}			
+	};
+
+	auto getWindow() { 
+
+		GLFWwindow* window;
+		window = glfwCreateWindow(resWidth, resHeight,				// Resolution
+			"Having problems with this one in constructor",         // Title
+			NULL,													// Monitor for fullscreen
+			NULL);													// Window to steal OpenGL context from - in most cases NULL
+
+		if (!window)
+		{
+			std::cout << "Failed to initialize GLFW window" << std::endl;
+			glfwTerminate();
+			//return 2; same as above
+		}
+
+		return window;
+
+	}
+
+	void opengl_intialize() {
```

Comment:
Do we need to call this explicitly, or can we do it for the user?

Reply:
We could add/move this to the constructor and have it called once its initialized?

Reply:
Or, we could make it a private method, which gets called in the constructor or before we do first action with drawing.

Reply:
I figured this out, now I understand what you mean but there is a problem.

If its initialized before glfwMakeContextCurrent(window); there is a bug and there is unexpected behavior.
For example the grid is not rendered and if you attempt to use the keyboard arrows for control you're gone...

Not sure how to explain this... its very strange.
Right now I'm trying to figure out a way to resolve this without messing out the code too much.

Reply:
In other words the opengl_intialize() needs to be called after glfwMakeContextCurrent

Reply:
Yes, exactly - not sooner